engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffQueryParameter.scala

```diff
@@ -2,19 +2,21 @@ package cromwell.engine.workflow.lifecycle.execution.callcaching
 
 import cats.data.{NonEmptyList, Validated}
 import cats.implicits._
+import cromwell.core.WorkflowId
+import lenthall.validation.ErrorOr.ShortCircuitingFlatMap
 import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheDiffQueryParameter.CallCacheDiffQueryCall
 
 import scala.util.{Failure, Success, Try}
 
 object CallCacheDiffQueryParameter {
-  case class CallCacheDiffQueryCall(workflowId: String, callFqn: String, jobIndex: Option[Int])
+  case class CallCacheDiffQueryCall(workflowId: WorkflowId, callFqn: String, jobIndex: Option[Int])
 
   private def missingWorkflowError(attribute: String) = s"missing $attribute query parameter".invalidNel
 
   def fromParameters(parameters: Seq[(String, String)]): Validated[NonEmptyList[String], CallCacheDiffQueryParameter] = {
     def extractIndex(parameter: String): Validated[NonEmptyList[String], Option[Int]] = {
       parameters.find(_._1 == parameter) match {
-        case Some((_, value)) => Try(value.toInt) match {
+        case Some((_, value)) => Try(value.trim.toInt) match {
```

Comment:
There is already `case None` below no ? Line 23 https://github.com/broadinstitute/cromwell/pull/2552/files#diff-418822f319fcfc8f92177448f49093f4R23